### PR TITLE
issue: 2667588 Add IPv6 MC

### PR DESCRIPTION
**Pull request by iftahl**
Add flags for IPv6 support:
    --mc-rx-if
    --mc-tx-if
They represent interface index, as integers.

Signed-off-by: Iftah Levi <iftahl@nvidia.com>

### CODE REVIEW

**Review by igor-ivanov — APPROVED** [review]

File: src/sockperf.cpp [hunk 1]
```diff
@@ -2801,11 +2855,20 @@ int sock_set_multicast(int fd, struct fds_data *p_data) {
...
-            //TODO; add IPv6 multicasting suport
+            log_err("For IPv6 - input must be an integer");
+            rc = SOCKPERF_ERR_UNSUPPORTED;
             break;
         }
     }
 
+    if (!rc && s_user_params.tx_mc_if_ix_specified) {
+        uint32_t if_ix = s_user_params.tx_mc_if_ix;
+        if (setsockopt(fd, IPPROTO_IPV6, IPV6_MULTICAST_IF, &if_ix, sizeof(int)) < 0) {
```
Comment (agalanin-at-nvidia) [inline src/sockperf.cpp:2866]: This code is probably not correct because `sizeof(int)` may differ from `sizeof(uint32)`

Comment (igor-ivanov) [reply src/sockperf.cpp:2866]: Agree, but it looks as minor issue as far our target arch x32 :) and x64

Comment (iftahl) [reply src/sockperf.cpp:2866]: agree with inconsistency, but sizeof(int) is always 4 bytes, so its safe.
sizeof(uint32_t) can vary - and this is a bigger issue.

Comment (agalanin-at-nvidia) [reply src/sockperf.cpp:2866]: probably you mean the opposite: sizeof(uint32_t)==4, sizeof(int) may vary [from 2 to 4](https://en.cppreference.com/w/cpp/language/types#Properties)

**Pull request merged**